hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+                price_decimals = Decimal(str(rule["priceDecimals"]))
+                quantity_decimals = Decimal(str(rule["amountDecimals"]))
+                # E.g. a price decimal of 2 means 0.01 incremental.
+                price_step = Decimal("1") / Decimal(str(math.pow(10, price_decimals)))
+                quantity_step = Decimal("1") / Decimal(str(math.pow(10, quantity_decimals)))
+                result[trading_pair] = TradingRule(trading_pair,
+                                                   min_price_increment=price_step,
+                                                   min_base_amount_increment=quantity_step)
+            except Exception:
+                self.logger().error(f"Error parsing the trading pair rule {rule}. Skipping.", exc_info=True)
+        return result
+
+    async def _api_request(self,
+                           method: str,
+                           path_url: str,
+                           params: Dict[str, Any] = {},
+                           is_auth_required: bool = False) -> Dict[str, Any]:
+        """
+        Sends an aiohttp request and waits for a response.
+        :param method: The HTTP method, e.g. get or post
+        :param path_url: The path url or the API end point
+        :param is_auth_required: Whether an authentication is required, when True the function will add encrypted
+        signature to the request.
+        :returns A response in json format.
+        """
+        url = f"{Constants.REST_URL}/{path_url}"
+        client = await self._http_client()
+        if is_auth_required:
+            # request_id = btc_markets_utils.RequestId.generate_request_id()
+            # data = {"params": params}
+            post_data = {"body": params}
+
+            if post_data.get("body", {}).get("side"):
+                if post_data['body']['side'] == 'BUY':
+                    post_data['body']['side'] = 'Bid'
+                elif post_data['body']['side'] == 'SELL':
+                    post_data['body']['side'] = 'Ask'
+
+            if method.upper() == "POST":
+                sign_params = self._btc_markets_auth.generate_auth_dict(method, path_url,
+                                                                        btc_markets_utils.get_ms_timestamp(),
+                                                                        post_data)
+            else:
+                sign_params = self._btc_markets_auth.generate_auth_dict(method, path_url,
+                                                                        btc_markets_utils.get_ms_timestamp(),
+                                                                        params)
+
+            headers = {
+                "Accept": "application/json",
+                "Accept-Charset": "UTF-8",
+                "Content-Type": "application/json",
+                "BM-AUTH-APIKEY": sign_params.get('api_key'),
+                "BM-AUTH-TIMESTAMP": str(sign_params.get('nonce')),
```

Comment:
If the timestamp is important for the authentication then the connector should be implemented using a TimeSynchronizer, as Binance spot connector.